java/maven/test/unit/src/org/netbeans/modules/maven/output/TestOutputListenerProviderTest.java

```diff
@@ -126,9 +129,179 @@ public void testFailfafeSeparateTestOuput() {
+        provider.processLine("[INFO] Surefire report directory: /target/surefire-reports", visitor);
+        assertNull(visitor.getOutputListener());
+        assertEquals(provider.outputDir, "/target/surefire-reports");
+        visitor.resetVisitor();
+
+        provider.processLine("Running com.example.SampleTest", visitor);
+        assertEquals(provider.runningTestClass, "com.example.SampleTest");
+        assertNull(visitor.getOutputListener());
+        visitor.resetVisitor();
+
+        // Failed tests.
+        provider.processLine("Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.003 sec <<< FAILURE! - in com.example.SampleTest", visitor);
+        // there must be a listener to navigate to test reports
+        assertNotNull(visitor.getOutputListener());
+        visitor.resetVisitor();
+
+        // Successful tests.
+        provider.processLine("Tests run: 1, Failures: 0, Errors: 0, Skipped: 0", visitor);
+        // there must be a listener to navigate to test reports
+        assertNull(visitor.getOutputListener());
+        visitor.resetVisitor();
         provider.sequenceFail("mojo-execute#surefire:test", visitor);
+    }
+
+    /**
+     * Test based on the output generated by Surefire 2.22.1 as well with 3.0.0-M3 with logging turned on
+     */
+    public void testSurefireTestOuput_2_22_later_withLogging() {
+        OutputVisitor visitor = new OutputVisitor();
+        visitor.resetVisitor();
+        provider.sequenceStart("mojo-execute#failsafe:integration-test", visitor);
+        assertNull(visitor.getOutputListener());
+        visitor.resetVisitor();
         
+        // Surefire report directory: /target/surefire-reports is not generated by default only with mvn -e
+        provider.processLine("[INFO] Surefire report directory: /target/surefire-reports", visitor);
+        assertNull(visitor.getOutputListener());
+        assertEquals(provider.outputDir, "/target/surefire-reports");
+        visitor.resetVisitor();
+
+        provider.processLine("[INFO] Running com.example.SampleTest", visitor);
+        assertEquals(provider.runningTestClass, "com.example.SampleTest");
+        assertNull(visitor.getOutputListener());
+        visitor.resetVisitor();
+
+        // Failed tests.
+        provider.processLine("[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.003 sec <<< FAILURE! - in com.example.SampleTest", visitor);
+        // there must be a listener to navigate to test reports
+        assertNotNull(visitor.getOutputListener());
+        visitor.resetVisitor();
+
+        // Successful tests.
+        provider.processLine("[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0", visitor);
+        // there must be a listener to navigate to test reports
+        assertNull(visitor.getOutputListener());
+        visitor.resetVisitor();
+        provider.sequenceFail("mojo-execute#surefire:test", visitor);
     }
-    
-}
+}
```

Comment:
You should add a new line.

Reply:
Is that "new line" part of code formatting convention? If yes can it be found somewhere and can I import it in NB IDE to do a "format code" that follows the conventions then?

Reply:
Maybe, I don't think we have an official rule at the moment, but the existing file has the new line at least. (Personally, I prefer it.)

Reply:
OK. Usually I use a new line a the very end of my sources files. Here, I just overlooked it. Shall I fix it?

Reply:
Yes, thank you. It seems that you added it to another file.